File: server/app/schemas/note.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class NoteBase(BaseModel):
    title: Optional[str] = None
    content: Optional[str] = None
    memo: Optional[str] = None
    is_favorite: Optional[bool] = None
    notebook: Optional[str] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None


class NoteCreate(NoteBase):
    title: str
    content: str
    notebook: str


class NoteUpdate(NoteBase):
    pass


class NoteInDBBase(BaseModel):
    id: Optional[UUID] = None
    title: Optional[str] = None
    content: Optional[str] = None
    user_id: Optional[UUID] = None
    notebook_id: Optional[UUID] = None
    summary: Optional[str] = None
    memo: Optional[str] = None
    is_favorite: Optional[bool] = None
    latitude: Optional[float] = None
    longitude: Optional[float] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        orm_mode = True


class NoteInDB(NoteInDBBase):
    id: UUID
    title: str
    content: str
    user_id: UUID
    notebook_id: UUID
    is_favorite: bool
    created_at: datetime
    updated_at: datetime


class Note(NoteInDBBase):
    pass

File: app/src/main/java/kr/ac/deu/se/wisenote/vo/note/Note.java
package kr.ac.deu.se.wisenote.vo.note;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class Note {
  @SerializedName("id")
  private String id;

  @SerializedName("title")
  private String title;

  @SerializedName("content")
  private String content;

  @SerializedName("user_id")
  private String user_id;

  @SerializedName("notebook_id")
  private String notebook_id;

  @SerializedName("summary")
  private String summary;

  @SerializedName("memo")
  private String memo;

  @SerializedName("notebook")
  private String notebook;

  @SerializedName("is_favorite")
  private Boolean is_favorite;

  @SerializedName("created_at")
  private Date created_at;

  @SerializedName("updated_at")
  private Date updated_at;

  public String getId() {
    return id;
  }

  public String getTitle() {
    return title;
  }

  public void setTitle(String title) {
    this.title = title;
  }

  public String getContent() {
    return content;
  }

  public String getUser_id() {
    return user_id;
  }

  public String getNotebook_id() {
    return notebook_id;
  }

  public void setNotebook_id(String notebook_id) {
    this.notebook_id = notebook_id;
  }

  public String getSummary() {
    return summary;
  }

  public String getMemo() {
    return memo;
  }

  public void setMemo(String memo) {
    this.memo = memo;
  }

  public Boolean getIs_favorite() {
    return is_favorite;
  }

  public Date getCreated_at() {
    return created_at;
  }

  public Date getUpdated_at() {
    return updated_at;
  }

  public String getNotebook() {
    return notebook;
  }

  public void setNotebook(String notebook) {
    this.notebook = notebook;
  }
}
